gnoland/website/pkgs/doc/types.go

```diff
@@ -0,0 +1,165 @@
+package doc
+
+import (
+	"strings"
+)
+
+type Package struct {
+	ImportPath string
+	Name       string
+	Doc        string
+	Filenames  []string
+	Funcs      []*Func
+	Methods    []*Func
+	Vars       []*Value
+	Consts     []*Value
+	Types      []*Type
+	Examples   []*Example
+}
+
+func (p *Package) populateType() {
+	p.populateTypeWithMethods()
+	p.populateTypeWithFuncs()
+	p.populateTypeWithValue()
```

Comment:
The way I see it, we should do an iteration over `p.Types` in this function, and then have

```go
t.Funcs, t.Methods = p.filterTypeFuncs(t.Name)
t.Consts, t.Vars = p.filterTypeValues(t.Name)
```

The helper functions remove funcs, consts and vars from the package and add them to local slices.

Reply:
I agree with @thehowl 💯%

These methods you've written down not only modify `Package`, but also modify the underlying member types at the same time - they have a lot of side-effects that are really not clear when you look at the name `populate`

Reply:
Completely agree